### ISSUE TITLE
Better handling of ARC-related property names

### ISSUE THREAD

**Issue by hunganh0403** [realm/realm-cocoa#4394]
## Goals

Report a brutal bug.

## Expected Results

Not Crash

## Actual Results

Crash

## Steps to Reproduce

Add data contain unicode
Get that unicode data --> crash

## Code Sample
````Swift
import Cocoa
import RealmSwift

@NSApplicationMain
class AppDelegate: NSObject, NSApplicationDelegate { //EXC_BAD_ACCESS at this line

    @IBOutlet weak var window: NSWindow!
    
    func applicationDidFinishLaunching(_ aNotification: Notification) {
        // Insert code here to initialize your application
        Database.add()
        _ = Database.get()
    }
}

class Database {
    static var realm: Realm {
        return try! Realm()
    }
    
    static func add() {
        try! realm.write {
            realm.add(NamerObject())
        }
    }
    
    static func get() -> String? {
        return realm.objects(NamerObject.self).first?.newName
    }
}

class NamerObject: Object {
    dynamic var newName: String = "tên mới"
}
````

If you want to test yourself, complete project here:
[RealmUnicode.zip](https://github.com/realm/realm-cocoa/files/623984/RealmUnicode.zip)

## Version of Realm and Tooling

Realm version: Newest 
(i just add github "realm/realm-cocoa" in my Cartfile and run carthage update this morning, nothing else)

Xcode version: Version 8.1 (8B62)

Swift version: 3.0.1

OSX: 10.11.6 (15G1108)

Pretty sure it happens with swift 2.3 xcode 8.0 too. I doubt that this happen in many platforms and versions

## What cause the bug: 

Property name begins with "new*" like "newA, newB, newXXX"

## How to fix

change 
````Swift    
dynamic var newName: String = "tên mới"
````
to 
````Swift    
dynamic var name: String = "tên mới" // or do whatever that not use "new*" for property name
````

## Broke-in-tear comment:

It took me tons of time to figure out why it crash while there're no clue about it, i have to cut of my code many, many time to the version that i reported.
Hope Realm team can fix or at least warn other about this because bug like this is so brutal, it might cause nightmare to other (like what happen to me)

**Comment by stel** [comment]
Thank you for reporting this issue and for the example project, @hunganh0403! I was also able to reproduce it, the crash happens during the autorelease pool draining. With memory management diagnostics turned on it seems that the resulting string was deallocated somewhere outside the autorelease pool. 

```
*** -[CFString release]: message sent to deallocated instance 0x60004213bfe0
```

/cc @bdash, @tgoyne

**Comment by tgoyne** [comment]
[ARC assigns special meaning to selectors starting with `new`, `alloc`, `copy`, `mutableCopy`, and `init`](http://clang.llvm.org/docs/AutomaticReferenceCounting.html#arc-method-families), which applies to `dynamic` properties in Swift. We should probably either reject properties with such names or try to return a non-autoreleased object from them.

**Comment by stel** [comment]
The issue also depends on the default value for this property: if some other string is specified the issue doesn't reproduce.

**Comment by austinzheng** [comment]
I believe this was fixed with 2.4.3. From the release notes:

```
* Reject Swift properties with names which cause them to fall in to ARC method
  families rather than crashing when they are accessed.
```

Closing this ticket.
